### ISSUE TITLE
Documentation of threshold_img unclear

### ISSUE THREAD

**Issue by JohannesWiesner** [nilearn/nilearn#2048]
1.) From the documentation of `threshold_img` it's unclear whether features below threshold will be kept or just set to zero. When I first saw the documentation I thought that `threshold_img` is a feature reduction procedure and will return a feature reduced subset. I had to find out by myself that features below threshold will be set to zero. The used phrase _"will be kept"_ also leads to the false idea that `threshold_img` is a feature reduction procedure.

2.) If I read the source code right `img_data[np.abs(img_data) < cutoff_threshold] = 0.)`, all features below threshold will be set to zero. This logically implies that all other values greater **or equal** to threshold will be kept. Therefore, the sentence 

> _The voxels which have intensities greater than this score will be kept._ 

should be changed to

>  _The voxels which have intensities greater or equal than this score will be kept._

3.) It's very nice that `threshold_img` can also handle percentiles because otherwise you always have to inspect the data first (or look at descriptive stats) to find a good voxel intensity value as threshold. However it would be nice to give an explanation of how the percentile rank is calculated behind the scenes. Again if I read the source code right the given percentile string (e.g. '80%') is converted to a percentile rank using `scipy.stats.scoreatpercentile`. It would be nice to give a reference to this function and to document which keyword-arguments of `scipy.stats.scoreatpercentile` are used when `check_threshold` is called.

**Comment by bthirion** [comment]
Thx for the feedback. I guess than nobody can improve this better than you ;-) 
Would you open a PR ?
I'm just a bit unsure about your last point: I don't think that the docstring should be too technical.

**Comment by dangom** [comment]
Concerning `threshold_img`, is it also a design decision for it to mutate the data instead of returning a copy, in contrast to what the other `image` functions do?

```python
from nilearn import datasets, image
mni = datasets.load_mni152_brain_mask()
image.threshold_img(mni, 1000)
# mni is now thresholded.
```

This should probably be documented as well.

**Comment by GaelVaroquaux** [comment]
> Concerning threshold_img, is it also a design decision for it to mutate the
> data instead of returning a copy, in contrast to what the other image functions
> do?

No, that's a bug. There should be a "copy=False" argument to control this
behavior.

**Comment by dangom** [comment]
I'd have thought so. Opening #2091

**Comment by Remi-Gau** [comment]
going through the backlog

I believe this can be closed given the fixing PR was merged.

**Comment by ymzayek** [comment]
@Remi-Gau I think the original issue about the docstring of `threshold_img` being unclear has not been addressed. (It seems that another issue was brought up under this and then resolved). If we look at the docstring here https://github.com/nilearn/nilearn/blob/main/nilearn/image/image.py#L863-L871 we can see that the 3 points in the original post have not been taken into account. Would you be interested in opening a PR to improve the docstring?

**Comment by Remi-Gau** [comment]
Will give it a go.
